extensions/kafka-streams/runtime/src/main/java/io/quarkus/kafka/streams/runtime/KafkaStreamsProducer.java

```diff
@@ -101,33 +106,53 @@ public KafkaStreamsProducer(KafkaStreamsSupport kafkaStreamsSupport, KafkaStream
                 runtimeConfig);
         this.kafkaAdminClient = Admin.create(getAdminClientConfig(kafkaStreamsProperties));
 
-        this.executorService = Executors.newSingleThreadExecutor();
+        this.executorService = executorService;
 
-        this.kafkaStreams = initializeKafkaStreams(kafkaStreamsProperties, runtimeConfig, kafkaAdminClient, topology.get(),
-                kafkaClientSupplier, stateListener, globalStateRestoreListener, uncaughtExceptionHandlerListener,
-                executorService);
+        this.topicsTimeout = runtimeConfig.topicsTimeout;
+        this.trimmedTopics = runtimeConfig.getTrimmedTopics();
+        this.streamsConfig = new StreamsConfig(kafkaStreamsProperties);
+        this.kafkaStreams = initializeKafkaStreams(streamsConfig, topology.get(),
+                kafkaClientSupplier, stateListener, globalStateRestoreListener, uncaughtExceptionHandlerListener);
         this.kafkaStreamsTopologyManager = new KafkaStreamsTopologyManager(kafkaAdminClient);
     }
 
-    @PostConstruct
-    public void postConstruct() {
+    public void onStartup(@Observes StartupEvent event) {
         if (kafkaStreams != null) {
             Arc.container().beanManager().getEvent().select(KafkaStreams.class).fire(kafkaStreams);
```

Comment:
You can replace `Arc.container().beanManager().getEvent().select(KafkaStreams.class)` with a method parameter of type `Event<KafkaStreams>`.

Reply:
Do you mean injecting `Event<KafkaStreams>` and fire in the method?

Reply:
I mean: 
```java
 public void onStartup(@Observes StartupEvent event, Event<KafaStreams> streamsEvent) {
     if (kafkaStreams != null) {
        steams.fire(kafkaStreams);
     }    
    // ...
 }
```

because parameters of observer methods are injection points...

Reply:
Yes, that's exactly what I had in mind, Thanks!